src/lib/x509/certstor.h

```diff
@@ -31,6 +31,15 @@ class BOTAN_DLL Certificate_Store
          find_cert(const X509_DN& subject_dn, const std::vector<byte>& key_id) const = 0;
 
       /**
+      * Find a certificate by searching for one with a matching SHA-1 hash of
+      * public key. Used for OCSP.
+      * @param key_hash SHA-1 hash of the subject's public key
+      * @return a matching certificate or nullptr otherwise
+      */
+      virtual std::shared_ptr<const X509_Certificate>
+         find_cert_by_pubkey_sha1(const std::vector<byte>& key_hash) const = 0;
```

Comment:
Unfortunately SHA-1 seems to be baked into OCSP especially in the key hash which is used to identify the response signer. Even in the latest OCSP RFC this is SHA-1 only (no algo id possible) and many responders on the net prefer providing a key hash instead of a DN. I don't see any way of implementing (a useful amount of) OCSP without SHA-1.